src/mesh/voxel_cloud.rs

```diff
@@ -141,6 +142,169 @@ impl VoxelCloud {
+            .map(|current_vc_bounding_box| {
+                // Make a bounding box of the source bounding box's mesh volume.
+                // This will be the volume to be scanned for any voxels.
+
+                // New voxel cloud that can encompass the source
+                // voxel cloud's mesh.
+                let mut target_voxel_cloud = VoxelCloud::from_cartesian_bounding_box(
+                    &current_vc_bounding_box,
+                    &voxel_dimensions,
+                );
+
+                for (one_dimensional, voxel) in target_voxel_cloud.voxel_map.iter_mut().enumerate()
+                {
+                    let cartesian_coordinate = one_dimensional_to_cartesian_coordinate(
+                        one_dimensional,
+                        &target_voxel_cloud.block_start,
+                        &target_voxel_cloud.block_dimensions,
+                        &target_voxel_cloud.voxel_dimensions,
+                    )
+                    .expect("Index out of bounds");
+
+                    // Set the new voxel state according to a sampled value from
+                    // the source voxel cloud.
+                    *voxel = source_voxel_cloud
+                        .voxel_at_cartesian_coords(&cartesian_coordinate)
+                        .unwrap_or(false);
+                }
+
+                target_voxel_cloud
+            })
+    }
+
+    /// Creates a new voxel cloud from another voxel cloud transformed (scaled,
+    /// rotated, moved) in a cartesian space.
+    /// # Panics
+    /// Panics if any of the voxel dimensions is below or equal to zero.
+    pub fn from_voxel_cloud_transformed(
+        source_voxel_cloud: &VoxelCloud,
+        voxel_dimension: f32,
+        cartesian_translation: &Vector3<f32>,
+        rotation_degrees: &(f32, f32, f32),
+        scale: &(f32, f32, f32),
+    ) -> Option<Self> {
+        if approx::relative_eq!(cartesian_translation, &Vector3::zeros())
+            && approx::relative_eq!(rotation_degrees.0, 0.0)
+            && approx::relative_eq!(rotation_degrees.1, 0.0)
+            && approx::relative_eq!(rotation_degrees.2, 0.0)
+            && approx::relative_eq!(scale.0, 1.0)
+            && approx::relative_eq!(scale.1, 1.0)
+            && approx::relative_eq!(scale.2, 1.0)
+        {
+            return VoxelCloud::from_voxel_cloud(
+                source_voxel_cloud,
+                &Vector3::new(voxel_dimension, voxel_dimension, voxel_dimension),
+            );
+        }
+
+        assert!(
+            voxel_dimension > 0.0,
+            "Voxel dimension is below or equal to zero"
+        );
+
+        source_voxel_cloud
+            .mesh_volume_bounding_box_cartesian()
+            .map(|current_vc_bounding_box| {
+                // FIXME: Heterogenous voxels require non-trivial compensation
+                // of final voxel cloud position after rotating if the volume
+                // equilibrium is not at the world origin.
```

Comment:
I'd be alright with having just cubic voxels. Let's leave the FIXME for now. We can disallow noncubic voxels in another PR.

Reply:
I'd keep the non-cubic voxel wherever we can and only drop them here.

Reply:
i'd keep it as is - cubic voxels here, non-cubic anywhere else